Improve the readability of the ip route show command

eoinaokane opened the pull request:
Signed-off-by: Eoin O'Kane <eoinandrewokane@gmail.com>

eoinaokane committed:
Improve the readability of the ip route show command

Signed-off-by: Eoin O'Kane <eoinandrewokane@gmail.com>

eoinaokane committed:
Improve the readability of the ip route show command

Signed-off-by: Eoin O'Kane <eoinandrewokane@gmail.com>

insomniacslk commented:
did you forget to include some files? I get errors because "hextoipaddress" and "flog" are missing. Also... flog?

eoinaokane commented:
> did you forget to include some files? I get errors because "hextoipaddress" and "flog" are missing. Also... flog?

Seem to have lost a function, if I want to update this pull request, let me see if I can amend this PR without nuking it.

eoinaokane committed:
Merge branch 'ip_route' of github.com:eoinaokane/u-root into ip_route

eoinaokane commented:
I think I have fixed it.

eoinaokane commented:
> did you forget to include some files? 

Seems it got nuked, added it back on

> I get errors because "hextoipaddress" and "flog" are missing. 

Fixed

> Also... flog?

Old school Pascal programming muscle memory

insomniacslk commented:
looking at this one again - it should be squashed because one of the three commits has merge conflicts

insomniacslk commented on cmds/core/ip/ip.go:
```diff
@@ -214,10 +216,125 @@ func routeshow() error {
+		eof := false
+
+		// Skip the top line, (already verified with regex)
+		for v :=1; v < len(rows) && !eof; v++ {
```
it's better to use a `bufio.Scanner` here, https://golang.org/pkg/bufio/#Scanner

insomniacslk commented on cmds/core/ip/ip.go:
```diff
@@ -214,10 +216,125 @@ func routeshow() error {
 	return nil
 }
 
+func hextoipaddress(src_address_hex string) (string, error) {
```
the whole function could be simplified to https://play.golang.org/p/oNZzRiqBsvM . Then the `net.IP` can be converted to string in the caller with the `.String` method of `net.IP`

insomniacslk commented:
I realized that this change is much more complicated than I initially thought. This is not just about parsing /proc/net/{ipv6_,}route but it requires using rtnetlink, which provides all the information we need. Discussion on https://github.com/u-root/u-root/issues/1369
